Core/include/Acts/TrackFinder/detail/VoidTrackFinderComponents.hpp

```diff
@@ -0,0 +1,38 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2018 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+#include <limits>
+
+#include "Acts/EventData/SourceLinkConcept.hpp"
+#include "Acts/EventData/TrackParameters.hpp"
+#include "Acts/Geometry/GeometryID.hpp"
+#include "Acts/Utilities/Logger.hpp"
+#include "Acts/Utilities/Result.hpp"
+#include "Acts/Utilities/TypeTraits.hpp"
+
+namespace Acts {
+///
+/// @brief void branch stopper
+struct VoidBranchStopper {
```

Comment:
Other than this noop one, do you have a "real" one anywhere? What would a branch stopper stop on, for example?

Reply:
No, I have not implemented a real one since the definition of a bad branch is quite tricky. But in general, the stopper could stop a branch based on something like the number of measurements/holes/outliers etc. (as those info stopped in `TipState`).